### ISSUE TITLE
Move to pyenv as virtual env manager

### ISSUE THREAD

**Issue by debasishdebs** [JanusGraph/janusgraph-python#13]
Currently, the library uses `virtualenv` package to create virtual environments inside which the library is build and installed to global python version.

The drawback with `virtualenv` is that it expects the Python interpretor to be installed in system and one of following commands work:
- python3
- python3.4
- python3.5
- python3.6

Since it expects Python to be installed, some users might experience Build failures. Also, we don't want users to be manually specifying the python version, as that should be developer knowledge.

To work around that, we are going to use `pyenv` which doesn't expect you to install a particular version of python, rather it compiles them, and creates an env. This will give us the flexibility of building the lib against each version of python (irrespective of it is being installed or not) and also test it at same time doesn't expect user to test which of above commands work, but still build without fails.

**Comment by debasishdebs** [comment]
Per comments from @chupman , we have decided that we won't Automatically install Python if that isn't present in system. We will simply throw and error, mentioning that the specified version of Python isn't installed. 

Per that conversation, I'll be adding a script namely install-python.sh which will take care of downloading and installing specified Python to make the job easier for Developer to install python, but at no point during Build will that script be invoked. 

The docs also needs to be updated according to same.

**Comment by debasishdebs** [comment]
**Update**: Though we have a script named install-python.sh, it will be removed once #14 is merged. We have also created 2 separate scripts for building on CentOS, Mac, Ubuntu, Windows.

build.sh expected to work on CentOS, Ubuntu, Mac, I've tested the same on CentOS and Ubuntu, while I'll have discussion with @chupman to test it on his Mac.
build.cmd works on Windows, and is tested against Win 7. Testing for Win 8, 10 & XP are welcome. (XP is optional)
